InformationScripting/src/query_prompt/QueryBuilder.cpp

```diff
@@ -89,7 +95,14 @@ Query* QueryBuilder::visitOperator(QueryBuilder* self, OperatorQueryNode* op)
 	else if (leftComposite)
 	{
 		// union
-		connectQueriesWith(composite, leftComposite, new UnionOperator(), right);
+		Query* unionQuery = nullptr;
+		if (op->op() == OperatorQueryNode::OperatorTypes::Pipe)
+			unionQuery = composite->addQuery(std::unique_ptr<Query>(new UnionOperator()));
```

Comment:
Oh, I totally misunderstood the purpose of this operator. Sorry for the confusion, just keep the current name, it is indeed as precise as it gets and is the right thing at this level.